go/common/crypto/signature/signature.go

```diff
@@ -609,33 +609,33 @@ func BuildPublicKeyBlacklist(allowTestKeys bool) {
 			blacklistedPublicKeys.Store(k, v)
 			return true
 		})
-	}
 
-	// Explicitly forbid other keys here.
-
-	// Small order points.
-	// See: https://github.com/jedisct1/libsodium/blob/master/src/libsodium/crypto_core/ed25519/ref10/ed25519_ref10.c#L1019.
-	for _, v := range []string{
-		// 0 (order 4).
-		"0000000000000000000000000000000000000000000000000000000000000000",
-		// 1 (order 1).
-		"0100000000000000000000000000000000000000000000000000000000000000",
-		// 2707385501144840649318225287225658788936804267575313519463743609750303402022 (order 8).
-		"26e8958fc2b227b045c3f489f2ef98f0d5dfac05d3c63339b13802886d53fc05",
-		// 55188659117513257062467267217118295137698188065244968500265048394206261417927 (order 8).
-		"c7176a703d4dd84fba3c0b760d10670f2a2053fa2c39ccc64ec7fd7792ac037a",
-		// p-1 (order 2).
-		"ecffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff7f",
-		// p (=0, order 4).
-		"edffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff7f",
-		// p+1 (=1, order 1).
-		"eeffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff7f",
-	} {
-		var pk PublicKey
-		if err := pk.UnmarshalHex(v); err != nil {
-			panic(err)
-		}
+		// Explicitly forbid other keys here.
```

Comment:
The small order points should unconditionally be rejected.

I assume you did this because you want to use `IsValid`.  There are certain things (namedly runtime IDs) that are currently `signature.PublicKey`, that should not be.  The solution here is not to accept invalid public keys in the public key verification routine, but to fix the types to stop expressing things as public keys when they are not (See #1693).

Reply:
Yeah, `IsValid` was failing on all-zeros runtime IDs.  I've removed the `IsValid` checks for runtime IDs and reverted the change in this file -- once #1693 is done and all runtime IDs are changed to be of type `common.Namespace`, those checks wouldn't have made any sense anyways :)